### PR TITLE
Tree:Fix ROOT-10829. Avoid using deleted object.

### PR DESCRIPTION
**Pull request by pcanal**
When deleting a TTree, we delete the list of friends and aliases.  Their deletion is broadcasted
through RecursiveRemove which eventually reached back to the original TTree.
And if TTree::RecursiveRemove then tries to call RecursiveRemove on those deleted list ... then
disaster follows ... in the form of:

```
pure virtual method called
terminate called without an active exception
Aborted (core dumped)
```

### CODE REVIEW

**Comment by pcanal** [comment]
The error on mac1014/python3 seems unrelated (missing roottest build product ```roottest/scripts/recordtiming_cc.so```

@phsft-bot build just on mac1014/python3

**Comment by pcanal** [comment]
@phsft-bot build just on mac1014/python3

**Comment by pcanal** [comment]
@phsft-bot build just on mac1014/python3

**Comment by pcanal** [comment]
@phsft-bot build just on mac1014/python3

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

**Comment by pcanal** [comment]
The errors are unrelated (remote file connection failures).

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: tree/tree/src/TTree.cxx [hunk 1]
```diff
@@ -7762,10 +7762,14 @@ void TTree::RecursiveRemove(TObject *obj)
    if (fTreeIndex == obj) {
       fTreeIndex = 0;
    }
-   if (fAliases) {
+   if (fAliases == obj) {
+      fAliases = 0;
```
Comment (Axel-Naumann) [inline tree/tree/src/TTree.cxx:7766]:
```suggestion
      fAliases = nullptr;
```

File: tree/tree/src/TTree.cxx [hunk 2]
```diff
@@ -7762,10 +7762,14 @@ void TTree::RecursiveRemove(TObject *obj)
...
+   } else if (fAliases) {
       fAliases->RecursiveRemove(obj);
    }
-   if (fFriends) {
+   if (fFriends == obj) {
+      fFriends = 0;
```
Comment (Axel-Naumann) [inline tree/tree/src/TTree.cxx:7771]:
```suggestion
      fFriends = nullptr;
```

**Comment by pcanal** [comment]
The error is pre-existing.

**Pull request merged**